### PR TITLE
Micro-optimize SUNMatrixWrapper::transpose

### PR DESCRIPTION
**Pull request by dweindl**
Quick test with a large model (nx=3665) shows up to 20% reduction in cpu_time and cpu_timeB

### CODE REVIEW

File: src/sundials_matrix_wrapper.cpp [hunk 1]
```diff
@@ -651,40 +652,38 @@ void SUNMatrixWrapper::transpose(SUNMatrixWrapper &C, const realtype alpha,
...
-            for (aidx = get_indexptr(acol);
-                 aidx < get_indexptr(acol+1); aidx++) {
-                widx = (acol/blocksize)*blocksize + get_indexval(aidx) % blocksize;
-                w.at(widx)++;
-                assert(w[widx] <= rows());
+        for (sunindextype acol = 0; acol < nrows; acol++) { /* row counts */
+            auto next_indexptr = get_indexptr(acol+1);
+            for (sunindextype aidx = get_indexptr(acol);
+                 aidx < next_indexptr; aidx++) {
+                sunindextype widx = (acol/blocksize)*blocksize + get_indexval(aidx) % blocksize;
```
Comment (dweindl) [inline src/sundials_matrix_wrapper.cpp:663]: @FFroehlich: I was wondering if all those `* blocksize / blocksize` are helpful here in any way

Comment (dweindl) [reply src/sundials_matrix_wrapper.cpp:663]: Ah, all integers. Forget it.

**Review by FFroehlich — APPROVED** [review]
👍  wonder whether it makes sense to replace other `.at(...)` in the same way.

**Comment by dweindl** [comment]
> wonder whether it makes sense to replace other `.at(...)` in the same way.

In several places it would, yes.

**Pull request merged**